### ISSUE TITLE
Some sort of settings system

### ISSUE THREAD

**Issue by AKuederle** [learn-and-code/MoneyPy#15]
Some sort of settings system, that allows the user to store some baseline configurations in a file.

- [ ] Figure out where things like that are stored on each platform
- [ ]  Create a convenient Singelton like Settings "class"/module that can be used everywhere

**Comment by tweakimp** [comment]
In what language should we hold the settings data? Json format?

**Comment by AKuederle** [comment]
Something we need to decide. Python has native support for json, ini
formats, or you can just use pure python files.

Also reminds me, it should be possible to layer the settings. Meaning user
settings overwrite base settings, project settings overwrite user settings

**Comment by BlitzKraft** [comment]
If we use python files, it can be achieved using sequential imports. However, one setback I can see here is that if one variable is declared in one of the files, and a conflicting option is set in another file, there might be unpredictable behavior. I might be overthinking this, but I want to use simple python files for ease of use. Especially, they can be edited by hand easily. 

My gripe is with json, where a forgotten comma could throw an error.

**Comment by AKuederle** [comment]
You actually want to allow conflicting keys. The important part is to
establish a hierarchy.
Python files might be the easiest ones for now. The only issue with them in
the long run, is that they allow arbitrary code execution, which you might
not want to have in a settings file. But for now I think we should go with
it.

**Comment by BlitzKraft** [comment]
I agree with a hierarchy, but the edge I was thinking about was something like this:

Say user file has `DARK = True`, and base setting has `BRIGHT = True` - this *could* lead to a complicated logic tree. If the variable names were same, I agree, the hierarchy will work as expected.

**Comment by AKuederle** [comment]
Aah I see. Of course you need to restrict the allowed set of keys. Or
rather there will only be one of the two keys in your example that is
actually read bei the program
